backend/api/models/User.py

```diff
@@ -96,7 +96,11 @@ def roles(self):
         """
         Roles applied to the User
         """
-        return Role.objects.filter(user_roles__user_id=self.id)
+        role = cache.get(f'role-user-{self.id}')
```

Comment:
I don't think we should cache user roles because if an admin changes the assigned roles, they won't be updated immediately in the cache and could cause confusion.